File: source/code/vbnet/authentication/create_directory_google.vb
Imports Stormpath.SDK.Provider

Dim googleDirectory = client.Instantiate(Of IDirectory)()
googleDirectory.SetName("My Google Directory")

Await client.CreateDirectoryAsync(
    googleDirectory,
    Function(options) options.ForProvider(client _
                                          .Providers() _
                                          .Google() _
                                          .Builder() _
                                          .SetClientId("YOUR_GOOGLE_CLIENT_ID") _
                                          .SetClientSecret("YOUR_GOOGLE_CLIENT_SECRET") _
                                          .SetRedirectUri("YOUR_GOOGLE_REDIRECT_URI") _
                                          .Build()))

File: source/code/nodejs/authentication/create_directory_google.js
var googleDirectory = {
  name : 'my-google-directory',
  description : 'A Google directory',
  provider: {
    providerId: 'google',
    clientId: 'YOUR_GOOGLE_CLIENT_ID',
    clientSecret: 'YOUR_GOOGLE_CLIENT_SECRET',
    redirectUri: 'YOUR_GOOGLE_REDIRECT_URI'
  }
};

client.createDirectory(googleDirectory, function (err, directory) {
  if (err) {
    return console.error(err);
  }

  console.log('Google directory created!');
});